.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md

```diff
@@ -0,0 +1,55 @@
+## Description
+
+This PR introduces a new <!-- insert new version here --> release in tezos-packaging.
+
+<!-- Depending on the type of the release some of the points below can be ommited.
+
+E.g. in case we introduce a new changes to our native packages, it's not necessary to bump
+used Tezos sources, create a new release in this repository and update brew formulas.
+
+Feel free to cross out points that aren't needed in some specific case.
+
+Note that opam-repository is usually updated with some delay, so CI may fail
+due to lack of updated opam Tezos packages. In such case feel free to comment
+out native-packaging-related CI steps from the pipeline until the opam-repository
+is updated.
+-->
+
+### Things to do in the release PR
+
+- [ ] Update Tezos sources and opam-repository (if needed) revisions in [sources.json](../../tree/master/nix/nix/sources.json).
+- [ ] Remove old bottles hashes from the brew formulas in [Formula directory](../../tree/master/Formula).
+- [ ] Update versions in brew formulas.
+- [ ] Update release number in [meta.json](../../tree/master/meta.json).
+
+#### In case the new Tezos release provides a new protocol and corresponding testnet
+
+- [ ] Support new protocol in [protocols.json](../../tree/master/protocols.json).
+- [ ] Support new protocol and testnet in native packaging.
+- [ ] Support new protocol and testnet in brew formulas.
+- [ ] Add tests for the new protocol and testnet.
+
+### Things to do after the release PR is merged
+
```

Comment:
I suggest adding a comment here specifying that these can be checked in this same PR after the fact.

But maybe that's clear already, so I'll let you choose.